### PR TITLE
v1.0.3

### PR DESCRIPTION
**Pull request by boxdrop**
Added support for adding insurance to shipments, change carrier, plus bugfixes

### CODE REVIEW

File: lib/BoxdropHelper.class.php [hunk 1]
```diff
@@ -257,4 +263,43 @@ public static function explodeString($delimiter, $input)
...
+	
+	        if ((int) $number == $test)
+	          $return = $test;
+	
+	      }
+	      else if ($has_dot === false && $has_comma !== false)
+	        $return = str_replace(',', '.', $return);
+	
+	      return (float) $return;
+	    }
```
Comment (Quetzacoalt91) [inline lib/BoxdropHelper.class.php]: You should also try this function to check if it fits your needs:

``` php
    public static function convertFormNumberToFloat($number)
    {
        return (float)str_replace(array('.', ','), array('','.'), $return);
    }
```

File: boxdropshipment.php [hunk 2]
```diff
@@ -143,7 +143,8 @@ public function install()
 				return false;
 			}
 
-			if (!$this->registerHook('actionValidateOrder') || !$this->registerHook('displayCarrierList') || !$this->registerHook('updateCarrier'))
+			if (!$this->registerHook('actionValidateOrder') || !$this->registerHook('displayCarrierList') ||
+			!$this->registerHook('updateCarrier') || !$this->registerHook('displayAdminOrderContentShip'))
```
Comment (Quetzacoalt91) [inline boxdropshipment.php]: Adding the new hook registration would be good in the upgrade file too.

File: upgrades/Upgrade-1.0.2.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * 
+ * @author sweber <sw@boxdrop.com>
+ * @param  mixed  $object
+ * @return boolean
+ */
+function upgrade_module_1_0_3()
+{
+	return Db::getInstance()->execute('ALTER TABLE `'._DB_PREFIX_.'boxdrop_order_shipment` ADD
+	 `insurance_amount` DECIMAL(7,2) NOT NULL DEFAULT \'0.00\' 
+	 AFTER `airwaybill`;');
```
Comment (Quetzacoalt91) [inline upgrades/Upgrade-1.0.2.php]: You did not update the CREATE TABLE request in _sql/create.php_, so this field will be missing on a fresh install.

File: lib/BoxdropHelper.class.php [hunk 4]
```diff
@@ -45,10 +47,15 @@ public static function getAdminDirPath()
...
+						if (file_exists($index_file))
 						{
-							$admin_dir = $absolute_element;
-							break;
+							$file_contents = file_get_contents($index_file);
+							if (strpos($file_contents, "define('_PS_ADMIN_DIR_', getcwd()") !== false)  
+							{								
+								$admin_dir = $absolute_element;
+								break;
+							}
```
Comment (Quetzacoalt91) [inline lib/BoxdropHelper.class.php:58]: I guess we could improve the way you create the admin environment for the next version of BoxDrop, in example with ModuleAdminController (and a ModuleFrontController for the front-office). When the controller will be called, you will be sure to be in an admin context, and all your checks won't be needed anymore. Please have a look on it, I'm sure your code will be safer and cleaner.

**Comment by boxdrop** [comment]
Thanks for you feedback, we have pushed the changes now.

**Comment by boxdrop** [comment]
comment.

**Comment by Quetzacoalt91** [comment]
Wouuups ! I have just seen that your upgrade folder is wrong ! It should be _upgrade_, not _upgrades_.

**Comment by boxdrop** [comment]
There you go!

**Comment by Quetzacoalt91** [comment]
Thanks @boxdrop, the pull-request has just been merged and the changes will be available from tomorrow.

Regards

**Pull request merged**